### PR TITLE
Initial SST integration.  

### PR DESCRIPTION
**Pull request by eisenhauer**
Open/Close functionality only.  One simple test included.

### CODE REVIEW

File: source/adios2/toolkit/sst/cp/marshal.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
+extern void adios_sst_write(SstStream s, void *BaseType, int BaseSize,
```
Comment (pnorbert) [inline source/adios2/toolkit/sst/cp/marshal.c]: Why is this extern?

**Comment by pnorbert** [comment]
[  9%] Building CXX object source/adios2/CMakeFiles/adios2.dir/core/IO.cpp.o
In file included from /home/adios/work/ADIOS2/source/adios2/core/IO.cpp:29:0:
/home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.h:17:17: fatal error: sst.h: No such file or directory

**Comment by pnorbert** [comment]
Some warnings:
[  4%] Building C object source/adios2/toolkit/sst/CMakeFiles/sst.dir/dp/dp.c.o
In file included from /home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp/dp.c:8:0:
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp_interface.h:175:43: warning: ‘struct _SstMetadata’ declared inside parameter list
                                           void **TimestepInfoPtr);
                                           ^
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp_interface.h:175:43: warning: its scope is only this definition or declaration, which is probably not what you want
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp_interface.h:175:43: warning: ‘struct _SstData’ declared inside parameter list
[  4%] Building C object source/adios2/toolkit/sst/CMakeFiles/sst.dir/dp/dummy_dp.c.o
In file included from /home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp/dummy_dp.c:10:0:
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp_interface.h:175:43: warning: ‘struct _SstMetadata’ declared inside parameter list
                                           void **TimestepInfoPtr);
                                           ^
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp_interface.h:175:43: warning: its scope is only this definition or declaration, which is probably not what you want
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp_interface.h:175:43: warning: ‘struct _SstData’ declared inside parameter list
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp/dummy_dp.c: In function ‘LoadDummyDP’:
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/dp/dummy_dp.c:594:38: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
     dummyDPInterface.provideTimestep = DummyProvideTimestep;

**Comment by eisenhauer** [comment]
Marshaling is not yet complete, but that function should be callable from the SST Engine.

**Comment by eisenhauer** [comment]
This, and the missing sst.h in your prior note, are probably a side effect of moving away from an independent SST that is not integrated.  Not noticed in my build environment because of files still around.  Let me fix.

More generally though, this passes all the CI tests, likely because it depends upon EVPath and doesn’t compile any of this if EVPath is not found.  We should probably think about how to deal with that.  I.E. include evpath as a third party library?  Have some CI environments with EVPath?

g

**Comment by pnorbert** [comment]
Still cannot build it.
[  6%] Building CXX object source/adios2/CMakeFiles/adios2.dir/engine/sst/SstReader.cpp.o
In file included from /home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.h:16:0,
                 from /home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.cpp:11:
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/sst.h:47:32: error: ‘MPI_Comm’ has not been declared
                                MPI_Comm comm);
                                ^
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/sst.h:56:32: error: ‘MPI_Comm’ has not been declared
                                MPI_Comm comm);
                                ^
/home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.cpp: In constructor ‘adios2::SstReader::SstReader(adios2::IO&, const string&, adios2::Mode, MPI_Comm)’:
/home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.cpp:22:30: error: ‘strcpy’ was not declared in this scope
     strcpy(cstr, name.c_str());
                              ^
/home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.cpp:24:48: error: invalid conversion from ‘MPI_Comm {aka ompi_communicator_t*}’ to ‘int’ [-fpermissive]
     m_Input = SstReaderOpen(cstr, NULL, mpiComm);
                                                ^
In file included from /home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.h:16:0,
                 from /home/adios/work/ADIOS2/source/adios2/engine/sst/SstReader.cpp:11:
/home/adios/work/ADIOS2/source/adios2/toolkit/sst/sst.h:55:18: note:   initializing argument 3 of ‘_SstStream* SstReaderOpen(const char*, const char*, int)’
 extern SstStream SstReaderOpen(const char *filename, const char *params,
                  ^

**Comment by eisenhauer** [comment]
Ugh.  I think I stupidly ran clang-format after testing.  That sometimes (helpfully) rearranges the order of include files...  Trying again.

**Comment by eisenhauer** [comment]
Hmm.  Still issues.  Going to pull this PR and resubmit shortly.

**Pull request closed without merging**